crates/rover-config/src/layout.rs

```diff
@@ -0,0 +1,21 @@
+use anyhow::{Error, Result};
+use std::env;
+use std::path::PathBuf;
+
+// ~/ or APOLLO_HOME
+//     .apollo/
+//         config.toml
```

Comment:
yeah so i toyed with this. if i am an end user.. i'm not gonna write `APOLLO_HOME = "./my_special_dir/.apollo"` this is why i think it should point at the parent

Reply:
to be clear:

> if we end up having multiple config directories (for example atlas) they should be in .apollo/atlas or .apollo/other-tool

i agree with that ^. i just don't think we should require end users to name the containing dir

Reply:
A user would set that only to override the default, right?
So if the default home would be `~/.apollo` and I want to override it, I'd do
`APOLLO_HOME = "/whatever/i/want` -- I'm suggesting `.apollo` as a top level, our code would not look for that string (".apollo") except if `APOLLO_HOME` is not defined, and so the user would not have to write it out themselves.

Reply:
i think it would be messy and increase the likelihood of a file collison.

if a user chose their own dir (let's say they do `APOLLO_HOME="./"`) like i do in the tests... this means that whatever we would otherwise put in `/.apollo` is now just.. sprayed all over `./`.

Reply:
(github is noting this is outdated, that's because i updated the comment becuase i'm not keeping this config in home, but rather the default config dir for the OS now, the comment was outdated)

Reply:
also, i removed by comment saying i agreed re: `.apollo/tool` .. if that's the case.. are you suggesting a namespace for this cli in the `.apollo` dir?

Reply:
> are you suggesting a namespace for this cli in the .apollo dir?

Good question. The decision depends on what we think is the foreseeable future. We know there's an `atlas` coming at some point, with some complicated config under it, so IMO it'd make sense to encapsulate it under its own folder, under `.apollo` -- I think I'm looking at `.apollo` as the true Apollo home on your fs, that would contain anything Apollo related.

Making structural changes in that folder are not as simple, since any decision to change/refactor would require at the very least some migration code, and in the worst case, would require users to do something manual (like move folders, change env vars, etc.) so in this case, I'd spend some time thinking about what is to come and try to do _some_ future proofing.

The current code really only contains the api key in a file under `.apollo` -- it's plausible that it's fine to keep there, if we think it's going to be used by more tools running on your machine that are not the CLI. I can't say what those would be yet. Maybe, for example, we'll make the VSCode extension look for an API key there?

When it comes to my expectations as a user from any tools "home" directory, I'd want _global_ settings defined under that folder to either not be nested, or be clearly nested under something that's obviously used by multiple tools.

For cli specific settings, that we know will not be used by other tools, it might make sense to nest those under `ap` or `rover` (i.e. whatever name we go with) now assuming that more folders for other tools will come.

> if a user chose their own dir (let's say they do APOLLO_HOME="./") ..

Come to think of it, what is the use case for letting users choose anything other than `~/.apollo`? I initially leaned towards giving users control, but I looked through some other CLI tools and couldn't find one that allows users to change this location.

Reply:
as the i write in the docs... default operating system directories are not always available in development environments, such as CI or a docker container, and/or they have locked down permissions. as a result, it can be helpful to let folks override because otherwise the command would *fail*. 

also the config is *not* stored in the home directory. it is stored in the OS default config dir.

Reply:
Right. Good point! Thanks for clarifying, sorry I missed that in the docs.

Reply:
some prior art:
- wrangler https://github.com/cloudflare/wrangler/blob/master/src/settings/global_config.rs#L7
- npm https://docs.npmjs.com/misc/config#npmrc-files
- cargo https://doc.rust-lang.org/cargo/reference/environment-variables.html

Reply:
Thanks those are really useful!

It looks like `CARGO_HOME` is exactly aligned with what I'm suggesting: the "home" implies `.cargo` itself. It defaults to `$HOME/.cargo` (or `%USERPROFILE%\.cargo` for win) but can be changed. Someone can choose to change it to `$HOME` -- that'd not be a great choice on their part.

`WRANGLER_HOME` is aligned with the current PR: the "home" implies the _parent_ for `.wrangler`.

npm is slightly different because it's a file, not a directory, it defaults to `$HOME/.npmrc` and uses `NPM_CONFIG_USERCONFIG` to set it, it implies the location of the file itself, not the parent of it.

This comes down to opinion, so I'm quite happy going with whatever you choose. ❤️

Reply:
latest commit addresses this!